tests/js/spec/views/issueList/overview.spec.jsx

```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
+        const projects = [
+          TestStubs.Project({
+            id: '1',
+            slug: 'foo',
+            isMember: true,
+            firstEvent: false,
+          }),
+          TestStubs.Project({
+            id: '2',
+            slug: 'bar',
+            isMember: true,
+            firstEvent: true,
+          }),
+          TestStubs.Project({
+            id: '3',
+            slug: 'baz',
+            isMember: true,
+            firstEvent: false,
+          }),
+        ];
+        MockApiClient.addMockResponse({
+          url: '/organizations/org-slug/sent-first-event/',
+          query: {
+            is_member: true,
+          },
+          body: {sentFirstEvent: true},
+        });
+        MockApiClient.addMockResponse({
+          url: '/organizations/org-slug/projects/',
+          body: projects,
+        });
+        wrapper = createWrapper({
+          organization: TestStubs.Organization({
+            projects,
+          }),
+        });
+        createWrapper({
+          organization: TestStubs.Organization({
+            projects,
+          }),
+        });
+
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
+
+        expect(wrapper.queryByTestId('awaiting-events')).not.toBeInTheDocument();
+      });
+
+      it('displays when all selected projects do not have first event', async function () {
+        const projects = [
+          TestStubs.Project({
+            id: '1',
+            slug: 'foo',
+            isMember: true,
+            firstEvent: false,
+          }),
+          TestStubs.Project({
+            id: '2',
+            slug: 'bar',
+            isMember: true,
+            firstEvent: false,
+          }),
+          TestStubs.Project({
+            id: '3',
+            slug: 'baz',
+            isMember: true,
+            firstEvent: false,
+          }),
+        ];
+        MockApiClient.addMockResponse({
+          url: '/organizations/org-slug/sent-first-event/',
+          query: {
+            project: [1, 2],
+          },
+          body: {sentFirstEvent: false},
+        });
+        MockApiClient.addMockResponse({
+          url: '/organizations/org-slug/projects/',
+          body: projects,
+        });
+        MockApiClient.addMockResponse({
+          url: '/projects/org-slug/foo/issues/',
+          body: [],
+        });
+
+        createWrapper({
+          selection: {
+            projects: [1, 2],
+            environments: [],
+            datetime: {period: '14d'},
+          },
+          organization: TestStubs.Organization({
+            projects,
+          }),
+        });
+
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
+
+        await waitFor(() =>
+          expect(wrapper.getByTestId('awaiting-events')).toBeInTheDocument()
```

Comment:
should we check for a rendered text instead a `data-test-id?` for instance `Waiting for events…`

Reply:
Why?

My assumption here is that querying by test id is more efficient than querying by text, but I haven't confirmed this is true.

Reply:
hm wouldn't it be better to test what information users see on the screen?

Reply:
> In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post "Making your UI tests resilient to change"

https://testing-library.com/docs/queries/bytestid/